File: app/src/main/java/com/example/cs_topics_project_test/function_java/Date.java
package com.example.cs_topics_project_test.function_java;

import androidx.annotation.NonNull;

public final class Date {
    private final int date;
    private final int month;
    private final int year;

    public Date(int date, int year, int month) {
        if (date > 31) {
            int mul = date / 31;
            month += mul;
            date = date % 31;
        }
        this.date = date;
        this.month = month;
        this.year = year;
    }

    public int getDate() {
        return date;
    }
    public int getMonth() {
        return month;
    }
    public int getYear() {
        return year;
    }

    @NonNull
    @Override
    public String toString() {
        return month + "/" + date + "/" + year;
    }
}

File: app/src/main/java/com/example/cs_topics_project_test/function/Date.kt
package com.example.cs_topics_project_test.function

class Date(year: Int, month: Int, date: Int) {
    private val date: Int
    private val month: Int
    private val year: Int

    init {
        this.year = year
        if (month > 12) throw IllegalArgumentException("invalid month as input")
        else this.month = month
        if (date > 31) throw IllegalArgumentException("invalid date as input")
        else this.date = date
    }

    fun getYear(): Int {
        return year
    }

    fun getMonth(): Int {
        return month
    }

    fun getDate(): Int {
        return date
    }

    override fun toString(): String {
        return "$month/$date/$year"
    }
}